src/Coven.Agents/Streaming/AgentStreamingOptions.cs

```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: BUSL-1.1
+namespace Coven.Agents.Streaming;
+
+public sealed class AgentStreamingOptions
+{
+    public bool Enabled { get; set; }
+
+    public IStreamSegmenter? Segmenter { get; set; }
```

Comment:
Shouldn't be nullable. Let's make sure this is only ever provided when streaming is enabled.